Storage/SQL/BrowserDB.swift

```diff
@@ -430,12 +430,20 @@ extension BrowserDB: Queryable {
 
     func queryReturnsResults(sql: String, args: Args?=nil) -> Deferred<Maybe<Bool>> {
         return self.runQuery(sql, args: args, factory: { row in true })
-         >>== { deferMaybe($0[0] ?? false) }
+            >>== { if ($0.status == .Success) && ($0.count > 0) {
+                return deferMaybe($0[0] ?? false)
```

Comment:
Also, what does $0[0] actually refer to? I'm assuming it's the first column of the current row pointed to by the cursor. If that is so, what does a boolean value of a column indicate?